napalm_nxos/nxos.py

```diff
@@ -369,7 +489,11 @@ def get_facts(self):
     def get_interfaces(self):
         interfaces = {}
         iface_cmd = 'show interface'
-        interfaces_out = self.device.show(iface_cmd)
+        if self.ssh_connection:
```

Comment:
I think this pattern is going to be a problem, it's very redundant and people if going to forget about it. I would certainly do ;)

I suggest hiding the logic inside some other method so you don't have to do the `if..else` when writing getters.